math/src/fft/fft_inputs.rs

```diff
@@ -0,0 +1,76 @@
+use super::StarkField;
+use crate::FieldElement;
+
+// FFTINPUTS TRAIT
+// ================================================================================================
+
+/// Defines the shape of the input passed as input for FFT computation.
+pub trait FftInputs<B: StarkField> {
+    /// Returns the length of this fftinputs.
+    fn len(&self) -> usize;
+
+    /// Combines the result of smaller discrete fourier transforms into a larger DFT.
+    fn butterfly(&mut self, offset: usize, stride: usize);
+
+    /// Combines the result of smaller discrete fourier transforms multiplied with a
+    /// twiddle factor into a larger DFT.
+    fn butterfly_twiddle(&mut self, twiddle: B, offset: usize, stride: usize);
+
+    /// Swaps the element at index i with the element at index j.
+    fn swap(&mut self, i: usize, j: usize);
+
+    /// Helper method used to replace each element in fftinputs with respective `poly`
+    /// item times `init_offset`, where the `init_offset` gets updated by `offset_factor`
+    /// right after each element update.
```

Comment:
This will need to be update after the change described above is made, but I'd also add concrete formulas for what this operation does.